### ISSUE TITLE
'use strict' not allowed in top-level modules when no-unused-expression is enabled

### ISSUE THREAD

**Issue by jbroni** [palantir/tslint#1115]
### Bug Report

- __TSLint version__: `3.7.3`
- __TypeScript version__: `1.8.9`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
namespace StrictTest {
    'use strict';
    export class StrictClass {
        test() {
            console.log('test');
        }
    }
}
```

with `tslint.json`:

```json
{
  "rules": {
    "no-unused-expression": true,
    "use-strict": [true, "check-module"]
  }
}
```

#### Actual behavior

```
Strict.ts[2, 5]: expected an assignment or function call
```
'use strict' expression triggers no-unused-expression violation

#### Expected behavior

'use strict' expression should not trigger no-unused-expression violation when placed in top of a module/namespace.
In particular, removing the 'use strict' expression from the above code causes a violation of the use-strict rule with the check-module argument.

**Comment by adidahiya** [comment]
Let's make sure to add a regression test for this kind of usage in https://github.com/palantir/tslint/blob/master/test/rules/no-unused-expression/test.ts.lint

**Comment by JKillian** [comment]
This is a bug, thanks for the heads-up! I guess then that there are four(?) valid locations for this in TS:

1. At the top of a file
2. At the top of a function/arrow function
3. At the top of a namespace/module
4. At the top of a class (?)

Looking at the [spec](https://tc39.github.io/ecma262/#sec-directive-prologues-and-the-use-strict-directive) I'm not entirely sure on #4, but since classes compile down to a function, might be logical?

**Comment by JKillian** [comment]
@jbroni I'll have this fixed and published before end of day today, and most likely much sooner

**Comment by jbroni** [comment]
@JKillian Thank you, I appreciate it!

**Comment by JKillian** [comment]
Fixed in the just published `v3.7.4` 🎉
